ratatui-core/src/symbols/merge.rs

```diff
@@ -0,0 +1,48 @@
+use crate::symbols::line::{BorderSymbol, LineStyle};
+
+/// Defines the merge strategy of overlapping characters.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Default)]
+pub enum MergeStyle {
+    /// Merges symbols only if an exact composite unicode character exists.
+    /// Example: `┐` and `┗` will be merged into `╄`
+    #[default]
+    Exact,
```

Comment:
Comment should cover a bit about why this is needed in a narrative form (e.g. There are some symbol combinations that are missing such as rounded corners combining with straight lines and thick lines with double lines.

I'd think in general that BestFit would be a better default, but given that I'm suggesting that Overwrite should be a variant, that makes more sense rather than having a two tier default (None / Some(Exact))